app/services/payment/spam_protection_service.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Payment
+  class SpamProtectionService
```

Comment:
I think we should rename this to `MultipleFailureChecker`
It is not really spam protection but multiple failure check

We have our strategy based on:

- ban if failed multiple times (timespan 3 days for example)
- ban if attempt to request multiple times (timespan 1 hour for example)